idaes/apps/grid_integration/tests/test_bidder.py

```diff
@@ -0,0 +1,207 @@
+        bidding_model_object = TestMissingModel(missing_attr=attr)
+        with pytest.raises(AttributeError, match=r".*{}().*".format(attr)):
+            bidder_object = Bidder(
+                bidding_model_object=bidding_model_object,
+                n_scenario=n_scenario,
+                solver=solver,
+                forecaster=forecaster,
+            )
+
+
+@pytest.mark.unit
+def test_n_scenario_checker():
+
+    solver = pyo.SolverFactory("cbc")
+    forecaster = TestingForecaster(horizon=horizon, n_sample=n_scenario)
+    bidding_model_object = TestingModel(horizon=horizon)
+
+    # test if bidder raise error when negative number of scenario is given
+    with pytest.raises(ValueError, match=r".*greater than zero.*"):
+        bidder_object = Bidder(
+            bidding_model_object=bidding_model_object,
+            n_scenario=-1,
+            solver=solver,
+            forecaster=forecaster,
+        )
+
+    # test if bidder raise error when float number of scenario is given
+    with pytest.raises(TypeError, match=r".*should be an integer.*"):
+        bidder_object = Bidder(
+            bidding_model_object=bidding_model_object,
+            n_scenario=3.0,
+            solver=solver,
+            forecaster=forecaster,
+        )
+
+
+@pytest.mark.unit
+def test_solver_checker():
+
+    solver = pyo.SolverFactory("cbc")
+    forecaster = TestingForecaster(horizon=horizon, n_sample=n_scenario)
+    bidding_model_object = TestingModel(horizon=horizon)
+
+    # test if bidder raise error when invalid solver is provided
+    invalid_solvers = [5, "cbc", "ipopt"]
+    for s in invalid_solvers:
+        with pytest.raises(TypeError, match=r".*not a valid Pyomo solver.*"):
+            bidder_object = Bidder(
+                bidding_model_object=bidding_model_object,
+                n_scenario=n_scenario,
+                solver=s,
+                forecaster=forecaster,
+            )
+
+
+@pytest.fixture
+def bidder_object():
+
+    solver = pyo.SolverFactory("cbc")
+    forecaster = TestingForecaster(horizon=horizon, n_sample=n_scenario)
+
+    # create a bidder model
+    bidding_model_object = TestingModel(horizon=horizon)
+    bidder_object = Bidder(
+        bidding_model_object=bidding_model_object,
+        n_scenario=n_scenario,
+        solver=solver,
+        forecaster=forecaster,
+    )
+    return bidder_object
+
+
+@pytest.mark.component
+def test_compute_bids(bidder_object):
+
+    marginal_cost = bidder_object.bidding_model_object.marginal_cost
+    gen = bidder_object.generator
+    default_bids = bidder_object.bidding_model_object.default_bids
+    pmin = bidder_object.bidding_model_object.pmin
+    pmax = bidder_object.bidding_model_object.pmax
+    date = "2021-08-20"
+
+    # test bidding when price forecast lower than marginal cost
+    # expect default bids
+    shift = 1
+    fixed_forecast = marginal_cost - shift
+    bids = bidder_object.compute_bids(date=date, hour=None, prediction=fixed_forecast)
```

Comment:
What happens if an invalid date is provided to `compute_bids`? Do we need a test for this resulting error message?